cprover_bindings/src/irep/irep_id.rs

```diff
@@ -847,14 +851,27 @@ impl ToString for IrepId {
         match self {
             IrepId::FreeformString(s) => return s.to_string(),
             IrepId::FreeformInteger(i) => return i.to_string(),
-            IrepId::FreeformHexInteger(i) => return format!("{:X}", i),
+            IrepId::FreeformHexInteger { value, width } => {
+                // CBMC expects two's complement for negative numbers.
+                // The bignum crate instead does sign/magnitude when making hex.
+                // So for negatives, do the two's complement ourselves.
+                if value.lt(&0u8.into()) {
+                    let one: BigInt = 1u8.into();
+                    let max: BigInt = one.clone().shl(width).sub(one.clone());
+                    assert!(value.abs().lt(&max));
+                    let two_complement = max.sub(value.abs().sub(one.clone()));
+                    return format!("{:X}", two_complement);
+                } else {
+                    return format!("{:X}", value);
```

Comment:
For a positive number, I'd like to check if its larger than can be represented. But, this would require knowing if the value is signed or unsigned. I'll see how much extra work that is.